### ISSUE TITLE
Influx server breaks down at random occasions, becoming unresponsive and spikes in memory and go routines

### ISSUE THREAD

**Issue by jgranstrom** [influxdata/influxdb#8440]
### Bug report

__System info:__
- InfluxDB docker image version 1.2.4 (non alpine)
- Running in docker swarm on docker version 17.05.0-ce
- Host machine is Ubuntu 16.04 (kernel 4.8)

__Steps to reproduce:__

I have not been able to find a consistent way to reproduce at this point, it happens at random occasions and does not correlate to any obvious changes to load or machine resources. It is generally 2-7 days between each occasion.

__Expected behavior:__
Server to keep being responsive at all times and not derail in resources.

__Actual behavior:__
The server suddenly becomes unresponsive, writes and queries all start to time out. The http API and CLI is responding, but times out for queries and writes. The memory usage and number of go routines climbs significantly until some point. After a few hours the server starts accepting writes again, but queries keep timing out. Restarting the server is required to get back into normal state. Stopping or restarting the docker container is not possible since the influx process becomes "defunct" and goes zombie after being killed. A full reboot of the machine is required to clean it up.

__Additional info:__
The server is not configured in any special way, everything is running with default settings.

We are writing about 4000 points per hour over about 2000 series in total in a single database and two measurements. Each serie has about 10 tags of which 3 are driving the series cardinality. Points are written in bulk at every 30 minutes to one of the measurements. A continuous query is running every 30 minutes, selecting from the first measurement and writing to the other. We currently have about 2M data points in the database.

These are the influx logs when things suddenly go wrong, writes and queries timing out.
[influx-timeouts.log](https://gist.github.com/jgranstrom/8c093993891d579a46e9587a6d7d1290)

Note how the writes go from responding 204 to 500 within the same set of writes, with a sudden high response time, and the continuous query start timing out. No other errors are being logged at this point.

I also attach logs from where the writes suddenly start coming back, but with the continuous query still timing out, several hours later. No server restarts have been made during this time.
[influx-partial-timeouts.log](https://gist.github.com/jgranstrom/00e3233970a35b7647791850d6e3966b)

Finally these are the logs from when the server was actually shut down manually. You can see a bunch of queries trying to run that never actually terminates, and after trying to stop the container it never actually stops and the influx process goes zombie. The queries in the logs runs without any issues under normal circumstances.
[influx-server-shutdown.log](https://gist.github.com/jgranstrom/5588f464b38982ec10147852bafaad1d)

I will include a few graphs of the influxdb internal metrics, as well as some reported metrics from the host itself. These clearly indicates that something is going on at the time of the incident. Note that the graphs are in a different time zone which is 2 hours ahead of the UTC logs, sorry about that. The anomalies in the graphs all start at the same time as errors are starting to pop up in the logs. The plunge at the end of each graph is due to server restart.

**Influx server usage**
<img width="1405" alt="screen shot 2017-05-29 at 16 54 28" src="https://cloud.githubusercontent.com/assets/3751473/26555958/0565a28c-4498-11e7-90d5-e844289783e3.png">

**Write timeouts**
<img width="1498" alt="screen shot 2017-05-29 at 16 53 00" src="https://cloud.githubusercontent.com/assets/3751473/26555887/964f2da0-4497-11e7-860e-a493e8f7b13c.png">

**Heap usage**
<img width="1504" alt="screen shot 2017-05-29 at 16 53 22" src="https://cloud.githubusercontent.com/assets/3751473/26555912/ca86a1a2-4497-11e7-995e-550a532345f7.png">

**Go routines**
<img width="1506" alt="screen shot 2017-05-29 at 16 53 34" src="https://cloud.githubusercontent.com/assets/3751473/26555919/d2fa73cc-4497-11e7-8c37-38d730f68e9e.png">

**Write fails**
*The writes fail for hours and then comes back in business*
<img width="1501" alt="screen shot 2017-05-29 at 16 53 48" src="https://cloud.githubusercontent.com/assets/3751473/26555933/e102a46c-4497-11e7-8806-c065b57ee3f1.png">

**Non terminating queries**
You can see the micro bumps overlapping the write timeouts that are actually the continuous query that is timing out, no other queries were running at that point. Incoming queries started at the end of the graphs and they were all timing out.
<img width="1499" alt="screen shot 2017-05-29 at 17 58 47" src="https://cloud.githubusercontent.com/assets/3751473/26556052/8b79e3ec-4498-11e7-9052-765cf5e4d29c.png">

It is worth noting that the host machine had well over 15GB of free memory during the full extent of the incident, CPUs were pretty much idle and 100+ GB of disk available.

Other details collected:
- There is no spike in system call errors
- No anomalies in open files / inodes / file I/O or similar is found at the time of the incident
- There is no write errors in the WAL
- There is no compaction errors
- There is no errors in tsm cache

I was unfortunately not able to get a profile at the time of this happening, but I'm attaching a profile from the restarted server in case it adds something that might be useful.
[influx-profile](https://gist.github.com/jgranstrom/44f9e3e4babcffc87d59c1d6dd7f9492)

**Comment by jwilder** [comment]
Can you attach the output of `show shards`?

**Comment by jgranstrom** [comment]
Definitely, here you go. `media_temp` and `test` databases are never queried or written to, and pretty much empty.

```
Connected to http://localhost:8086 version 1.2.4
InfluxDB shell version: 1.2.4
> show shards
name: _internal
id  database  retention_policy shard_group start_time           end_time             expiry_time          owners
--  --------  ---------------- ----------- ----------           --------             -----------          ------
126 _internal monitor          126         2017-05-24T00:00:00Z 2017-05-25T00:00:00Z 2017-06-01T00:00:00Z
127 _internal monitor          127         2017-05-25T00:00:00Z 2017-05-26T00:00:00Z 2017-06-02T00:00:00Z
128 _internal monitor          128         2017-05-26T00:00:00Z 2017-05-27T00:00:00Z 2017-06-03T00:00:00Z
130 _internal monitor          130         2017-05-27T00:00:00Z 2017-05-28T00:00:00Z 2017-06-04T00:00:00Z
131 _internal monitor          131         2017-05-28T00:00:00Z 2017-05-29T00:00:00Z 2017-06-05T00:00:00Z
132 _internal monitor          132         2017-05-29T00:00:00Z 2017-05-30T00:00:00Z 2017-06-06T00:00:00Z
135 _internal monitor          135         2017-05-30T00:00:00Z 2017-05-31T00:00:00Z 2017-06-07T00:00:00Z
136 _internal monitor          136         2017-05-31T00:00:00Z 2017-06-01T00:00:00Z 2017-06-08T00:00:00Z


name: media_metrics
id  database      retention_policy shard_group start_time           end_time             expiry_time          owners
--  --------      ---------------- ----------- ----------           --------             -----------          ------
4   media_metrics autogen          4           2017-02-06T00:00:00Z 2017-02-13T00:00:00Z 2017-02-13T00:00:00Z
8   media_metrics autogen          8           2017-02-13T00:00:00Z 2017-02-20T00:00:00Z 2017-02-20T00:00:00Z
16  media_metrics autogen          16          2017-02-20T00:00:00Z 2017-02-27T00:00:00Z 2017-02-27T00:00:00Z
24  media_metrics autogen          24          2017-02-27T00:00:00Z 2017-03-06T00:00:00Z 2017-03-06T00:00:00Z
32  media_metrics autogen          32          2017-03-06T00:00:00Z 2017-03-13T00:00:00Z 2017-03-13T00:00:00Z
40  media_metrics autogen          40          2017-03-13T00:00:00Z 2017-03-20T00:00:00Z 2017-03-20T00:00:00Z
48  media_metrics autogen          48          2017-03-20T00:00:00Z 2017-03-27T00:00:00Z 2017-03-27T00:00:00Z
61  media_metrics autogen          61          2017-03-27T00:00:00Z 2017-04-03T00:00:00Z 2017-04-03T00:00:00Z
69  media_metrics autogen          69          2017-04-03T00:00:00Z 2017-04-10T00:00:00Z 2017-04-10T00:00:00Z
77  media_metrics autogen          77          2017-04-10T00:00:00Z 2017-04-17T00:00:00Z 2017-04-17T00:00:00Z
85  media_metrics autogen          85          2017-04-17T00:00:00Z 2017-04-24T00:00:00Z 2017-04-24T00:00:00Z
93  media_metrics autogen          93          2017-04-24T00:00:00Z 2017-05-01T00:00:00Z 2017-05-01T00:00:00Z
100 media_metrics autogen          100         2017-05-01T00:00:00Z 2017-05-08T00:00:00Z 2017-05-08T00:00:00Z
108 media_metrics autogen          108         2017-05-08T00:00:00Z 2017-05-15T00:00:00Z 2017-05-15T00:00:00Z
116 media_metrics autogen          116         2017-05-15T00:00:00Z 2017-05-22T00:00:00Z 2017-05-22T00:00:00Z
124 media_metrics autogen          124         2017-05-22T00:00:00Z 2017-05-29T00:00:00Z 2017-05-29T00:00:00Z
133 media_metrics autogen          133         2017-05-29T00:00:00Z 2017-06-05T00:00:00Z 2017-06-05T00:00:00Z


name: media_temp
id database   retention_policy shard_group start_time           end_time             expiry_time          owners
-- --------   ---------------- ----------- ----------           --------             -----------          ------
54 media_temp autogen          54          2017-02-27T00:00:00Z 2017-03-06T00:00:00Z 2017-03-06T00:00:00Z
55 media_temp autogen          55          2017-03-06T00:00:00Z 2017-03-13T00:00:00Z 2017-03-13T00:00:00Z


name: test
id  database retention_policy shard_group start_time           end_time             expiry_time          owners
--  -------- ---------------- ----------- ----------           --------             -----------          ------
129 test     autogen          129         2017-05-22T00:00:00Z 2017-05-29T00:00:00Z 2017-05-29T00:00:00Z
134 test     autogen          134         2017-05-29T00:00:00Z 2017-06-05T00:00:00Z 2017-06-05T00:00:00Z
```

**Comment by jwilder** [comment]
Looks like the shard retention policy enforcement is causing lock contention when purging shards.  Are you able to test with the current nightly?

**Comment by jgranstrom** [comment]
@jwilder Alright, I should be able to test with nightly for a few days and see if it still appears. Do I need to be worried about any potential breaking changes going from 1.2.4 to nightly and later potentially going back?

**Comment by jgranstrom** [comment]
@jwilder  So here's another instance of the incident. This is still with 1.2.4 but I managed to get logs, stats, shards and a profile **during** the incident.

<img width="515" alt="screen shot 2017-06-02 at 12 01 00" src="https://cloud.githubusercontent.com/assets/3751473/26721198/580855c0-478b-11e7-834d-585a7a1c478f.png">
<img width="480" alt="screen shot 2017-06-02 at 12 01 16" src="https://cloud.githubusercontent.com/assets/3751473/26721202/5a3e6c80-478b-11e7-8b8f-6659391a50cd.png">


[all debug extracts](https://gist.github.com/jgranstrom/172a239c678e44e5909a4bcfc3b1d93e)

**Comment by jwilder** [comment]
@jgranstrom When this occurs, can you run `iostat -xz 1 30` as well?  It looks like writes to the wal were all blocking on disk IO/fsyncs (this time) which usually indicates the disks are saturated.  What kind of disks do you have and how are they configured within your container?

Were you able to test w/ 1.3?

**Comment by jgranstrom** [comment]
@jwilder I am currently testing with the nightly at `1.3.0~n201705270801` and after ~5 days there have been no issues. It is a bit early to say that we're good but I will keep running it for another week or 2 and report back.

We are running on fast SSD disks with sufficient storage and quite a lot of margin in terms I/O and so on. We are running on digital ocean which are quite well situated when it comes to disks. However, we are using [GlusterFS](https://www.gluster.org/) as a layer below the docker containers in order to have the services host agnostic to support docker swarm failover in case a host goes down. This means that the influx data is stored in a distributed filesystem that can in theory be affected by network related issues or congestion. We have used gluster for quite some time and with multiple different types of storage systems such as postgres/redis/elasticsearch/rabbitmq etc without issues, but maybe there is some compatibility challenges with how influx is writing/reading to disk that can be affected by this fact?

I will definitely run iostat and post back in case we run into the issue again.

Thank you for taking a look at it!

**Comment by jgranstrom** [comment]
@jwilder we just ran in to this against using the nightly at `1.3.0~n201705270801`.

Same exact symptoms and behavior recorded.

Here's the `show shards` again

```
Connected to http://localhost:8086 version 1.3.0~n201705270801
InfluxDB shell version: 1.3.0~n201705270801
> show shards
name: _internal
id  database  retention_policy shard_group start_time           end_time             expiry_time          owners
--  --------  ---------------- ----------- ----------           --------             -----------          ------
141 _internal monitor          141         2017-06-05T00:00:00Z 2017-06-06T00:00:00Z 2017-06-13T00:00:00Z
144 _internal monitor          144         2017-06-06T00:00:00Z 2017-06-07T00:00:00Z 2017-06-14T00:00:00Z
145 _internal monitor          145         2017-06-07T00:00:00Z 2017-06-08T00:00:00Z 2017-06-15T00:00:00Z
146 _internal monitor          146         2017-06-08T00:00:00Z 2017-06-09T00:00:00Z 2017-06-16T00:00:00Z
147 _internal monitor          147         2017-06-09T00:00:00Z 2017-06-10T00:00:00Z 2017-06-17T00:00:00Z
148 _internal monitor          148         2017-06-10T00:00:00Z 2017-06-11T00:00:00Z 2017-06-18T00:00:00Z
149 _internal monitor          149         2017-06-11T00:00:00Z 2017-06-12T00:00:00Z 2017-06-19T00:00:00Z
150 _internal monitor          150         2017-06-12T00:00:00Z 2017-06-13T00:00:00Z 2017-06-20T00:00:00Z

name: media_metrics
id  database      retention_policy shard_group start_time           end_time             expiry_time          owners
--  --------      ---------------- ----------- ----------           --------             -----------          ------
4   media_metrics autogen          4           2017-02-06T00:00:00Z 2017-02-13T00:00:00Z 2017-02-13T00:00:00Z
8   media_metrics autogen          8           2017-02-13T00:00:00Z 2017-02-20T00:00:00Z 2017-02-20T00:00:00Z
16  media_metrics autogen          16          2017-02-20T00:00:00Z 2017-02-27T00:00:00Z 2017-02-27T00:00:00Z
24  media_metrics autogen          24          2017-02-27T00:00:00Z 2017-03-06T00:00:00Z 2017-03-06T00:00:00Z
32  media_metrics autogen          32          2017-03-06T00:00:00Z 2017-03-13T00:00:00Z 2017-03-13T00:00:00Z
40  media_metrics autogen          40          2017-03-13T00:00:00Z 2017-03-20T00:00:00Z 2017-03-20T00:00:00Z
48  media_metrics autogen          48          2017-03-20T00:00:00Z 2017-03-27T00:00:00Z 2017-03-27T00:00:00Z
61  media_metrics autogen          61          2017-03-27T00:00:00Z 2017-04-03T00:00:00Z 2017-04-03T00:00:00Z
69  media_metrics autogen          69          2017-04-03T00:00:00Z 2017-04-10T00:00:00Z 2017-04-10T00:00:00Z
77  media_metrics autogen          77          2017-04-10T00:00:00Z 2017-04-17T00:00:00Z 2017-04-17T00:00:00Z
85  media_metrics autogen          85          2017-04-17T00:00:00Z 2017-04-24T00:00:00Z 2017-04-24T00:00:00Z
93  media_metrics autogen          93          2017-04-24T00:00:00Z 2017-05-01T00:00:00Z 2017-05-01T00:00:00Z
100 media_metrics autogen          100         2017-05-01T00:00:00Z 2017-05-08T00:00:00Z 2017-05-08T00:00:00Z
108 media_metrics autogen          108         2017-05-08T00:00:00Z 2017-05-15T00:00:00Z 2017-05-15T00:00:00Z
116 media_metrics autogen          116         2017-05-15T00:00:00Z 2017-05-22T00:00:00Z 2017-05-22T00:00:00Z
124 media_metrics autogen          124         2017-05-22T00:00:00Z 2017-05-29T00:00:00Z 2017-05-29T00:00:00Z
133 media_metrics autogen          133         2017-05-29T00:00:00Z 2017-06-05T00:00:00Z 2017-06-05T00:00:00Z
142 media_metrics autogen          142         2017-06-05T00:00:00Z 2017-06-12T00:00:00Z 2017-06-12T00:00:00Z
151 media_metrics autogen          151         2017-06-12T00:00:00Z 2017-06-19T00:00:00Z 2017-06-19T00:00:00Z

name: media_temp
id database   retention_policy shard_group start_time           end_time             expiry_time          owners
-- --------   ---------------- ----------- ----------           --------             -----------          ------
54 media_temp autogen          54          2017-02-27T00:00:00Z 2017-03-06T00:00:00Z 2017-03-06T00:00:00Z
55 media_temp autogen          55          2017-03-06T00:00:00Z 2017-03-13T00:00:00Z 2017-03-13T00:00:00Z

name: test
id  database retention_policy shard_group start_time           end_time             expiry_time          owners
--  -------- ---------------- ----------- ----------           --------             -----------          ------
129 test     autogen          129         2017-05-22T00:00:00Z 2017-05-29T00:00:00Z 2017-05-29T00:00:00Z
134 test     autogen          134         2017-05-29T00:00:00Z 2017-06-05T00:00:00Z 2017-06-05T00:00:00Z
143 test     autogen          143         2017-06-05T00:00:00Z 2017-06-12T00:00:00Z 2017-06-12T00:00:00Z
152 test     autogen          152         2017-06-12T00:00:00Z 2017-06-19T00:00:00Z 2017-06-19T00:00:00Z
```

`show diagnostics` is actually not responding, it just hangs.

Debug extracts can be viewed [here](https://gist.github.com/jgranstrom/a5d882b9a5a9b1952cff09dcad4d9822)

**Comment by jgranstrom** [comment]
@jwilder we are still experiencing this about 1-2 times a week. Here's an `iostat` at the time of the most recent incident while trying to run queries that are not terminating during the time of the `iostat` running. Do you have any more thought on where to dig further?

```
root@marvin:~# iostat -xz 1 30
Linux 4.8.0-53-generic (marvin)   06/18/2017  _x86_64_  (12 CPU)

avg-cpu:  %user   %nice %system %iowait  %steal   %idle
           2.17    0.00    1.25    0.14    0.08   96.36

Device:         rrqm/s   wrqm/s     r/s     w/s    rkB/s    wkB/s avgrq-sz avgqu-sz   await r_await w_await  svctm  %util
vda               0.00    81.05    1.85   72.14    48.17  2318.85    63.98     0.31    4.16    0.49    4.25   0.24   1.78

avg-cpu:  %user   %nice %system %iowait  %steal   %idle
           1.76    0.00    0.92    0.00    0.00   97.32

Device:         rrqm/s   wrqm/s     r/s     w/s    rkB/s    wkB/s avgrq-sz avgqu-sz   await r_await w_await  svctm  %util
vda               0.00     0.00    0.00    9.00     0.00    88.00    19.56     0.00    0.00    0.00    0.00   0.00   0.00

avg-cpu:  %user   %nice %system %iowait  %steal   %idle
           2.77    0.00    0.59    0.00    0.00   96.65

Device:         rrqm/s   wrqm/s     r/s     w/s    rkB/s    wkB/s avgrq-sz avgqu-sz   await r_await w_await  svctm  %util
vda               0.00     0.00    0.00    9.00     0.00    88.00    19.56     0.00    0.44    0.00    0.44   0.44   0.40

avg-cpu:  %user   %nice %system %iowait  %steal   %idle
           1.76    0.00    1.09    0.08    0.00   97.07

Device:         rrqm/s   wrqm/s     r/s     w/s    rkB/s    wkB/s avgrq-sz avgqu-sz   await r_await w_await  svctm  %util
vda               0.00   219.00    0.00   12.00     0.00   976.00   162.67     0.01    0.67    0.00    0.67   0.67   0.80

avg-cpu:  %user   %nice %system %iowait  %steal   %idle
           1.76    0.00    0.84    0.00    0.33   97.07

Device:         rrqm/s   wrqm/s     r/s     w/s    rkB/s    wkB/s avgrq-sz avgqu-sz   await r_await w_await  svctm  %util
vda               0.00     0.00    0.00    9.00     0.00    88.00    19.56     0.00    0.00    0.00    0.00   0.00   0.00

avg-cpu:  %user   %nice %system %iowait  %steal   %idle
           2.09    0.00    1.00    0.00    0.00   96.91

Device:         rrqm/s   wrqm/s     r/s     w/s    rkB/s    wkB/s avgrq-sz avgqu-sz   await r_await w_await  svctm  %util
vda               0.00     0.00    0.00    9.00     0.00    88.00    19.56     0.00    0.00    0.00    0.00   0.00   0.00

avg-cpu:  %user   %nice %system %iowait  %steal   %idle
           2.60    0.00    1.09    0.00    0.08   96.23

Device:         rrqm/s   wrqm/s     r/s     w/s    rkB/s    wkB/s avgrq-sz avgqu-sz   await r_await w_await  svctm  %util
vda               0.00     0.00    0.00    9.00     0.00    88.00    19.56     0.00    0.44    0.00    0.44   0.44   0.40

avg-cpu:  %user   %nice %system %iowait  %steal   %idle
           2.44    0.00    0.76    0.00    0.00   96.80

Device:         rrqm/s   wrqm/s     r/s     w/s    rkB/s    wkB/s avgrq-sz avgqu-sz   await r_await w_await  svctm  %util
vda               0.00     0.00    0.00    9.00     0.00    88.00    19.56     0.01    0.89    0.00    0.89   0.89   0.80

avg-cpu:  %user   %nice %system %iowait  %steal   %idle
           1.85    0.00    0.92    0.00    0.08   97.15

Device:         rrqm/s   wrqm/s     r/s     w/s    rkB/s    wkB/s avgrq-sz avgqu-sz   await r_await w_await  svctm  %util
vda               0.00   192.00    0.00   12.00     0.00   868.00   144.67     0.01    1.00    0.00    1.00   1.00   1.20

avg-cpu:  %user   %nice %system %iowait  %steal   %idle
           2.85    0.00    0.92    0.00    0.08   96.14

Device:         rrqm/s   wrqm/s     r/s     w/s    rkB/s    wkB/s avgrq-sz avgqu-sz   await r_await w_await  svctm  %util
vda               0.00     0.00    0.00    9.00     0.00    88.00    19.56     0.01    0.89    0.00    0.89   0.89   0.80

avg-cpu:  %user   %nice %system %iowait  %steal   %idle
           2.01    0.00    0.59    0.00    0.08   97.32

Device:         rrqm/s   wrqm/s     r/s     w/s    rkB/s    wkB/s avgrq-sz avgqu-sz   await r_await w_await  svctm  %util
vda               0.00     0.00    0.00    9.00     0.00    88.00    19.56     0.01    0.89    0.00    0.89   0.89   0.80

avg-cpu:  %user   %nice %system %iowait  %steal   %idle
           3.02    0.00    1.76    0.00    0.08   95.13

Device:         rrqm/s   wrqm/s     r/s     w/s    rkB/s    wkB/s avgrq-sz avgqu-sz   await r_await w_await  svctm  %util
vda               0.00     2.00    0.00   20.00     0.00   180.00    18.00     0.00    0.00    0.00    0.00   0.00   0.00

avg-cpu:  %user   %nice %system %iowait  %steal   %idle
           2.18    0.00    1.01    0.00    0.08   96.73

Device:         rrqm/s   wrqm/s     r/s     w/s    rkB/s    wkB/s avgrq-sz avgqu-sz   await r_await w_await  svctm  %util
vda               0.00     0.00    0.00   13.00     0.00   104.00    16.00     0.00    0.31    0.00    0.31   0.31   0.40

avg-cpu:  %user   %nice %system %iowait  %steal   %idle
           2.10    0.00    1.26    0.25    0.42   95.98

Device:         rrqm/s   wrqm/s     r/s     w/s    rkB/s    wkB/s avgrq-sz avgqu-sz   await r_await w_await  svctm  %util
vda               0.00   267.00    3.00   18.00    12.00  1212.00   116.57     0.03    1.52    5.33    0.89   1.14   2.40

avg-cpu:  %user   %nice %system %iowait  %steal   %idle
           2.94    0.00    0.76    0.00    0.00   96.31

Device:         rrqm/s   wrqm/s     r/s     w/s    rkB/s    wkB/s avgrq-sz avgqu-sz   await r_await w_await  svctm  %util
vda               0.00     0.00    0.00    9.00     0.00    88.00    19.56     0.01    0.89    0.00    0.89   0.89   0.80

avg-cpu:  %user   %nice %system %iowait  %steal   %idle
           2.68    0.00    0.59    0.00    0.00   96.73

Device:         rrqm/s   wrqm/s     r/s     w/s    rkB/s    wkB/s avgrq-sz avgqu-sz   await r_await w_await  svctm  %util
vda               0.00     0.00    0.00    9.00     0.00    88.00    19.56     0.00    0.00    0.00    0.00   0.00   0.00

avg-cpu:  %user   %nice %system %iowait  %steal   %idle
           1.92    0.00    1.09    0.08    0.17   96.74

Device:         rrqm/s   wrqm/s     r/s     w/s    rkB/s    wkB/s avgrq-sz avgqu-sz   await r_await w_await  svctm  %util
vda               0.00    38.00    0.00  808.00     0.00  3436.00     8.50     2.48    3.06    0.00    3.06   0.03   2.40

avg-cpu:  %user   %nice %system %iowait  %steal   %idle
           2.09    0.00    0.67    0.00    0.08   97.15

Device:         rrqm/s   wrqm/s     r/s     w/s    rkB/s    wkB/s avgrq-sz avgqu-sz   await r_await w_await  svctm  %util
vda               0.00     0.00    0.00    9.00     0.00    88.00    19.56     0.00    0.44    0.00    0.44   0.44   0.40

avg-cpu:  %user   %nice %system %iowait  %steal   %idle
           2.34    0.00    0.84    0.00    0.08   96.74

Device:         rrqm/s   wrqm/s     r/s     w/s    rkB/s    wkB/s avgrq-sz avgqu-sz   await r_await w_await  svctm  %util
vda               0.00   192.00    0.00   12.00     0.00   868.00   144.67     0.01    1.00    0.00    1.00   0.67   0.80

avg-cpu:  %user   %nice %system %iowait  %steal   %idle
           1.68    0.00    1.01    0.00    0.00   97.32

Device:         rrqm/s   wrqm/s     r/s     w/s    rkB/s    wkB/s avgrq-sz avgqu-sz   await r_await w_await  svctm  %util
vda               0.00     0.00    0.00    9.00     0.00    88.00    19.56     0.01    0.89    0.00    0.89   0.89   0.80

avg-cpu:  %user   %nice %system %iowait  %steal   %idle
           2.18    0.00    0.50    0.00    0.08   97.24

Device:         rrqm/s   wrqm/s     r/s     w/s    rkB/s    wkB/s avgrq-sz avgqu-sz   await r_await w_await  svctm  %util
vda               0.00     0.00    0.00    9.00     0.00    88.00    19.56     0.00    0.44    0.00    0.44   0.44   0.40

avg-cpu:  %user   %nice %system %iowait  %steal   %idle
           1.84    0.00    0.84    0.00    0.08   97.24

Device:         rrqm/s   wrqm/s     r/s     w/s    rkB/s    wkB/s avgrq-sz avgqu-sz   await r_await w_await  svctm  %util
vda               0.00     0.00    0.00   11.00     0.00    96.00    17.45     0.01    0.73    0.00    0.73   0.73   0.80

avg-cpu:  %user   %nice %system %iowait  %steal   %idle
           2.76    0.00    0.84    0.00    0.08   96.32

Device:         rrqm/s   wrqm/s     r/s     w/s    rkB/s    wkB/s avgrq-sz avgqu-sz   await r_await w_await  svctm  %util
vda               0.00     0.00    0.00    9.00     0.00    88.00    19.56     0.00    0.44    0.00    0.44   0.44   0.40

avg-cpu:  %user   %nice %system %iowait  %steal   %idle
           2.26    0.00    1.42    0.00    0.00   96.31

Device:         rrqm/s   wrqm/s     r/s     w/s    rkB/s    wkB/s avgrq-sz avgqu-sz   await r_await w_await  svctm  %util
vda               0.00   188.00    0.00   12.00     0.00   852.00   142.00     0.00    0.00    0.00    0.00   0.00   0.00

avg-cpu:  %user   %nice %system %iowait  %steal   %idle
           2.50    0.00    1.09    0.00    0.08   96.33

Device:         rrqm/s   wrqm/s     r/s     w/s    rkB/s    wkB/s avgrq-sz avgqu-sz   await r_await w_await  svctm  %util
vda               0.00     0.00    0.00    9.00     0.00    88.00    19.56     0.01    1.33    0.00    1.33   1.33   1.20

avg-cpu:  %user   %nice %system %iowait  %steal   %idle
           1.59    0.00    0.67    0.00    0.00   97.74

Device:         rrqm/s   wrqm/s     r/s     w/s    rkB/s    wkB/s avgrq-sz avgqu-sz   await r_await w_await  svctm  %util
vda               0.00     0.00    0.00    9.00     0.00    88.00    19.56     0.01    0.89    0.00    0.89   0.89   0.80

avg-cpu:  %user   %nice %system %iowait  %steal   %idle
           2.09    0.00    1.00    0.00    0.33   96.57

Device:         rrqm/s   wrqm/s     r/s     w/s    rkB/s    wkB/s avgrq-sz avgqu-sz   await r_await w_await  svctm  %util
vda               0.00     0.00    0.00   93.00     0.00   484.00    10.41     0.01    0.09    0.00    0.09   0.09   0.80

avg-cpu:  %user   %nice %system %iowait  %steal   %idle
           2.76    0.00    2.34    0.00    0.17   94.73

Device:         rrqm/s   wrqm/s     r/s     w/s    rkB/s    wkB/s avgrq-sz avgqu-sz   await r_await w_await  svctm  %util
vda               0.00     0.00    0.00    9.00     0.00    88.00    19.56     0.00    0.44    0.00    0.44   0.44   0.40

avg-cpu:  %user   %nice %system %iowait  %steal   %idle
           2.51    0.00    2.60    0.00    0.00   94.89

Device:         rrqm/s   wrqm/s     r/s     w/s    rkB/s    wkB/s avgrq-sz avgqu-sz   await r_await w_await  svctm  %util
vda               0.00   184.00    0.00   13.00     0.00   840.00   129.23     0.02    1.85    0.00    1.85   1.54   2.00

avg-cpu:  %user   %nice %system %iowait  %steal   %idle
           2.93    0.00    1.42    0.00    0.08   95.57

Device:         rrqm/s   wrqm/s     r/s     w/s    rkB/s    wkB/s avgrq-sz avgqu-sz   await r_await w_await  svctm  %util
vda               0.00     0.00    1.00   14.00     4.00   176.00    24.00     0.00    0.27    0.00    0.29   0.27   0.40
```

**Comment by daveselan** [comment]
We are seeing the exact same behaviour running nightly build since 25th May?  I was planning on re-installing an older version to verify this.

**Comment by jgranstrom** [comment]
@jwilder We are still having a lot of issues with this. We had this incident twice in two days the last two days. I am now running it without GlusterFS, and directly on a fast SSD to see if that is a factor. I will report back after a few days with the results. If there's anything else I can investigate I would be happy to hear it.

**Comment by daveselan** [comment]
We haven't had the opportunity to role back yet. We are bringing a new server online next week, so one of these problematic servers will be downgraded then.

We have continued to see this problem at an increased frequency, probably now several influx hangs a day.

As a matter of interest, we have another almost identical setup that's been running fine for 2 months. The main setup differences are SSD with InfluxDB memory indexing on disk ("tsi1") on the working server vs HDD with standard settings on the hanging server. The collected data isn't exactly the same, however, the schema and data rate are identical, and quantity of systems/metrics collected for are very similar.

FYI - Attached a screen capture (hope it's not too big) of the last 2 days showing some internal Influx metrics.
Not sure how useful it is, but you can see the much longer response times and erratic behaviour, compared to the stable server that's on the right.

![image](https://user-images.githubusercontent.com/20041843/27959313-d8fa2ace-631e-11e7-9e9b-5d4c25cf1861.png)

**Comment by jwilder** [comment]
This sounds very similar to #8500 which we are fixing for 1.3.

If you are running a nightly or build based off of 1.3, it would really helpful to grab profiles when you see the hang:

```
curl -o profiles.tar.gz "http://localhost:8086/debug/pprof/all?cpu=true"
```

In general, running with HDDs is not recommended.  You will not see the same performance as directly attached SSDs (which is the only configuration we recommend for production workloads).

**Comment by daveselan** [comment]
This hanging "deadlock" has gone away after we stopped updates to one of our databases...

So, we have two databases on each of our servers; one being very small compared to the main one. Updates for both were staggered to remove any overlap. On Friday I stopped updates to the smaller database and the server has been stable for 4 days, which hasn't been the case up to this point. I can give more details if needed.

FYI - in the hung state, any curl debug profiling commands would also hang and never return.
